xapian-letor/feature/feature.cc

```diff
@@ -36,10 +39,17 @@ void
 Feature::operator=(const Feature & o)
 {
     LOGCALL_VOID(API, "Feature::operator=", o);
-    internal = o.internal;
+    feature_db = o.feature_db;
+    feature_query = o.feature_query;
+    feature_doc = o.feature_doc;
+    termfreq = o.termfreq;
+    inverse_doc_freq = o.inverse_doc_freq;
+    doc_length = o.doc_length;
+    collection_length = o.collection_length;
+    collection_termfreq = o.collection_termfreq;
 }
```

Comment:
Indeed - pre-C++11, the way to suppress the default copy constructor and/or assignment operator is just to declare (but **NOT** define) private versions.

It's not a good idea to also define them, because even private versions could still be called by methods of the class itself, or by friend classes/functions).

C++11 provides a cleaner way to do this by declaring these methods with `= delete` (the main benefit is that the compiler can give a better error message, because with the old-way it doesn't know if the definition is missing or deliberately omitted).  However, we're currently avoiding using C++11-specific features in the public API headers because at the moment it's still common to have to specify special compiler options to enable C++11 mode; if we use C++11-specific features in the API headers, that forces the build systems of code using Xapian to have to arrange for those compiler-specific options to be passed.  So for now, please stick with the old way (which still works with C++11 too).